src/sacn/dmx_merger.c

```diff
@@ -974,6 +966,44 @@ MergerState* construct_merger_state(sacn_dmx_merger_t handle, const SacnDmxMerge
   return merger_state;
 }
 
+/*
+ * Obtains the state structures for the source and merger specified by the given handles.
+ *
+ * Keep in mind that merger_state or source_state can be NULL if only interested in one or the other.
+ *
+ * Call sacn_lock() before using this function or the state data.
+ */
+etcpal_error_t lookup_state(sacn_dmx_merger_t merger, sacn_source_id_t source, MergerState** merger_state,
+                            SourceState** source_state)
+{
+  etcpal_error_t result = kEtcPalErrOk;
+
+  MergerState* my_merger_state = NULL;
+  SourceState* my_source_state = NULL;
+
+  // Look up the merger state.
+  my_merger_state = etcpal_rbtree_find(&mergers, &merger);
+
+  if (!my_merger_state)
+    result = kEtcPalErrNotFound;
+
+  // Look up the source state.
+  if ((result == kEtcPalErrOk) && source_state)
+  {
+    my_source_state = etcpal_rbtree_find(&my_merger_state->source_state_lookup, &source);
+
+    if (!my_source_state)
+      result = kEtcPalErrNotFound;
+  }
+
+  if (merger_state)
```

Comment:
Since the function returns an error code if one of the two states is not found, I recommend not writing to either of the output pointers if one of the lookups fail:
```
if (result == kEtcPalErrOk)
{
  if (merger_state)
    *merger_state = my_merger_state;
  if (source_state)
    *source_state = my_source_state;
}
```